### PR TITLE
Reducer and action

### PR DESCRIPTION
**Pull request by mahdimahdawi**
**This PR Contains:**
-  Created Redux directory and placed `books.js` and `categories.js` inside each folder.
- Defined action types for adding and removing a book.
- Created the initial state with an empty array of books.
- Exported Action Creators for the actions.
-Added reducer and exported as default.
- Defined state changes to the actions that were created.
- In case of unknown action - return to the current state.
- Inside the  `Categories.js` file.
- Defined an action type to check the status.
- Set the initial state with an empty array of categories.
- Exported Action Creators for the actions.
- Created reducer and exported as default.
- For the check status action return a string "Under construction".
- Inside `Configure.js` file.
- Imported the necessary methods from Redux Toolkit.
-  Combined both reducers into a root reducer by using the configureStore function.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { configureStore } from '@reduxjs/toolkit';
+import reducer from './books/books';
+import catREducer from './categories/categories';
+
+export default configureStore({
+  reducer: {
+    books: reducer,
+    categories: catREducer,
+  },
+});
```
Comment (mateo951) [inline src/redux/configureStore.js:10]: Well done! configureStore.js is properly added 🔥

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (mateo951) [inline src/redux/books/books.js]: Defines `ADD_BOOK` and `REMOVE_BOOK` as action types ✔

Comment (mateo951) [reply src/redux/books/books.js]: Please be sure that your actions types follow the ducks pattern 🤓 



     const ADD_BOOK = bookstore/books/ADD_BOOK

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const CEHCK_STATUS = 'CEHCK_STATUS';
```
Comment (mateo951) [inline src/redux/categories/categories.js]: Defines `CHECK_STATUS` as action type ✔

Comment (mateo951) [reply src/redux/categories/categories.js]: As I mentioned before, please change it to follow the duck pattern 😉

File: src/redux/books/books.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  };
+}
+
+// Remove Book Action
+export function removeBook(id) {
+  return {
+    type: REMOVE_BOOK,
+    id,
+  };
+}
```
Comment (mateo951) [inline src/redux/books/books.js:39]: Creates and Exports **Action Creators**. (`addBook` and `removeBook` functions) 💪😎

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @mahdimahdawi, 

**🔥Highlights🔥**
- The structure of the files follows the ducks pattern 🙌

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: src/redux/books/books.js (hunk 4 above)
Comment (mahdimahdawi) [reply src/redux/books/books.js:39]: This part is not clear to me kindly guide.

**Comment by mahdimahdawi** [comment]
Hi Villagomez,
Hope you doing well.
Thank you for your nice review really appreciate your patience,
Unfortunately, I didn't catch your last comment clearly, If it is possible
kindly give me some more information about last comment.
Looking forward to hearing from you.
best regards.
Mahdi Mahdawi.

**Comment by mahdimahdawi** [comment]
**Required Changes:**
- Action types follow the duck's pattern.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @mahdimahdawi  

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Duck pattern well implemented
- Actions and Reducers ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 


tufoin


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**